hr_employee_seniority/models/hr_employee.py

```diff
@@ -22,29 +22,49 @@ class HrEmployee(models.Model):
         compute='_compute_months_service',
     )
 
-    @api.depends('contract_ids')
+    def _first_contract(self):
+        Contract = self.env['hr.contract']
+        return Contract.search([('employee_id', '=', self.id)],
+                               order='date_start asc', limit=1)
+
+    @api.depends('contract_ids', 'initial_employment_date')
     def _compute_months_service(self):
+        date_now = fields.Date.today()
```

Comment:
@leemannd The same issue with calculation, previous days in different month is not giving you full month, add this and try it:

def check_next_days(date1, date2):
        if date1.day == 1:
            days_in_month = calendar.monthrange(date2.year, date2.month)[1]
            if date2.day == days_in_month:
                return 1
        elif date1.day == date2.day + 1:
            return 1
        return 0